packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -51,23 +52,30 @@ export class SummarizeHeuristicData implements ISummarizeHeuristicData {
  */
 export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
     private readonly idleTimer: Timer;
+    private readonly minOpsForAttemptOnClose: number;
 
     public constructor(
         private readonly heuristicData: ISummarizeHeuristicData,
         private readonly configuration: ISummaryConfiguration,
         private readonly trySummarize: (reason: SummarizeReason) => void,
-        private readonly minOpsForAttemptOnClose = 50,
     ) {
+        assert(this.configuration.state !== "disabled" &&
+            this.configuration.state !== "disableHeuristics",
+            "Configuration state should not be disabled or disableHeuristics");
```

Comment:
Given that this class only works with enabled heuristics, it's better to push this requirement to caller by changing type of configuration from ISummaryConfiguration to ISummaryConfigurationHeuristicSettings